frontend/tests/playwright/product/cross_sell.spec.ts

```diff
@@ -59,14 +56,11 @@ test.describe('Cross-sell test', () => {
 
    test('Cross-sell products can be added to cart', async ({ page }) => {
       const products = await page.getByTestId('cross-sell-item');
-      const count = await products.count();
 
       const allProductTitles = [];
       let expectedTotalPrice = 0;
 
-      for (let i = 0; i < count; i++) {
-         const product = products.nth(i);
-
+      for (const product of await products.all()) {
```

Comment:
Hmm, will `product.all()` get executed once at just the start of the loop or every time we loop? 

I think we should set the result to a constant outside of the loop and then use evaluated constant to iterate on.

Reply:
I think there might be a misunderstanding with the `products` and `product` constants.

The `products` constant is defined on top of the test on line 58: 
 https://github.com/iFixit/react-commerce/blob/65ed8888f57297e94d2d2a8103763a002e043bda/frontend/tests/playwright/product/cross_sell.spec.ts#L58

We call `products.all()` to iterate for single `const product`.

I used the example in https://playwright.dev/docs/api/class-locator#locator-all.

Reply:
TIL!